src/lib/project.js

```diff
@@ -332,7 +339,42 @@ async function scaffoldNext() {
   }
 };`
   );
+  newNextConfig = newNextConfig.replace(
+    'reactStrictMode: true',
+    'reactStrictMode: false'
+  );
   fs.writeFileSync(path.join('ui', 'next.config.js'), newNextConfig);
+
+  const tsconfig = `
+    {
+      "compilerOptions": {
+        "target": "ES2019",
+        "module": "es2022",
+        "lib": ["dom", "dom.iterable", "esnext"],
+        "allowJs": true,
+        "skipLibCheck": true,
+        "strict": true,
+        "strictPropertyInitialization": false, // to enable generic constructors, e.g. on CircuitValue
+        "forceConsistentCasingInFileNames": true,
+        "noEmit": true,
+        "esModuleInterop": true,
+        "module": "esnext",
+        "moduleResolution": "node",
+        "experimentalDecorators": true,
+        "emitDecoratorMetadata": true,
+        "resolveJsonModule": true,
+        "isolatedModules": true,
+        "jsx": "preserve",
+        "incremental": true
+      },
+      "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
+      "exclude": ["node_modules"]
+    }
+  `;
+
```

Comment:
```
        "target": "ES2019", // es5 in contracts tsconfig
        "module": "es2022",// esnext in contracts tsconfig,
        "strictPropertyInitialization": false, // to enable generic constructors, e.g. on CircuitValue
       "experimentalDecorators": true,
        "emitDecoratorMetadata": true
```
These are the only compiler options that differ from the defaults generated tsconfig in the UI project. Were you able to confirm that each of these options were necessary to get everything working?

Reply:
yep. Without these, smart contracts fail to build (we're using typescript features that need these things there)

Reply:
Thanks for verifying.